### PR TITLE
Attr support

### PR DESCRIPTION
**Pull request by yunstanford**
i removed the json_schema part as we are going to use your new lib.

### CODE REVIEW

**Comment by yunstanford** [comment]
I think i need to think up a edge case which can break cattrs.unstructure. :)

File: ubuild.py [hunk 1]
```diff
@@ -10,6 +11,16 @@ def main(build):
     build.packages.install(".", develop=True)
     # we install flask to allow testing the example.
     build.packages.install("flask")
+    # the version of cattrs in pypi is very old.
+    # api has also been changed in master branch,
+    # let's install master branch
+    ROOT = os.path.dirname(sys.executable)
+    CATTR_MASTER_BRANCH = "https://github.com/Tinche/cattrs/tarball/master"
```
Comment (toumorokoshi) [inline ubuild.py]: you can install from git directly:

https://github.com/toumorokoshi/notebooks/blob/master/ubuild.py#L5

File: transmute_core/object_serializers/attrs_serializer.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    def __init__(self):
+        pass
+
+    def load(self, model, value):
+        """
+        Converts unstructured data into structured data, recursively.
+        """
+        try:
+            res = structure(value, model)
+        except Exception as e:
```
Comment (toumorokoshi) [inline transmute_core/object_serializers/attrs_serializer.py]: the specific exception should be caught here. a generic try/catch could catch other issues unrelated to the destructuring of the object.

File: transmute_core/object_serializers/attrs_serializer.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import attr
+
+
+class AttrsSerializer(ObjectSerializer):
+    """
+    An ObjectSerializer which allows the serialization of
+    basic types and attrs classes.
+    """
+
+    def __init__(self):
```
Comment (toumorokoshi) [inline transmute_core/object_serializers/attrs_serializer.py]: you don't need to add this method if it's a no-op.

File: transmute_core/object_serializers/attrs_serializer.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+          such as, int, boolean, dict, other classes.
+        """
+        try:
+            res = unstructure(value)
+        except Exception as e:
+            raise SerializationException(str(e))
+        return res
+
+
+# Help Functions
```
Comment (toumorokoshi) [inline transmute_core/object_serializers/attrs_serializer.py]: can we remove this comment?

File: transmute_core/object_serializers/attrs_serializer.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+from .interface import ObjectSerializer
+from cattr import structure, unstructure
+from ..exceptions import SerializationException
+from cattr.vendor.typing import (
```
Comment (toumorokoshi) [inline transmute_core/object_serializers/attrs_serializer.py]: are any of these imports used?

File: transmute_core/object_serializers/attrs_serializer.py (hunk 3 above)
Comment (yunstanford) [inline transmute_core/object_serializers/attrs_serializer.py]: i will remove

File: transmute_core/object_serializers/attrs_serializer.py (hunk 2 above)
Comment (yunstanford) [inline transmute_core/object_serializers/attrs_serializer.py]: make senses.

File: transmute_core/object_serializers/attrs_serializer.py (hunk 4 above)
Comment (yunstanford) [inline transmute_core/object_serializers/attrs_serializer.py]: sure.

File: transmute_core/object_serializers/attrs_serializer.py (hunk 5 above)
Comment (yunstanford) [inline transmute_core/object_serializers/attrs_serializer.py]: haha, it has been used for to_json_schema method before, but i have removed that part.

Comment (yunstanford) [inline transmute_core/object_serializers/attrs_serializer.py]: i will clean those up

**Comment by toumorokoshi** [comment]
Looks good, thanks! I don't want to merge it in while it relies on a cattrs library that's unavailable yet, so I'm putting in the attrs_support branch for now.

**Comment by toumorokoshi** [comment]
I've merged this code in manually. Thanks!

**Pull request closed without merging**